django/db/models/query.py

```diff
@@ -464,7 +464,41 @@ def _prepare_for_bulk_create(self, objs):
                 obj.pk = obj._meta.pk.get_pk_value_on_save(obj)
             obj._prepare_related_fields_for_save(operation_name='bulk_create')
 
-    def bulk_create(self, objs, batch_size=None, ignore_conflicts=False):
+    def _select_on_conflict(
+        self, ignore_conflicts, update_conflicts, update_fields, unique_fields
+    ):
+        if ignore_conflicts and update_conflicts:
+            raise ValueError(
+                'You can only assign one conflicts plan, ignore_conflicts or update_conflicts'
+            )
+        if ignore_conflicts:
+            self._check_on_conflicts_supported(OnConflict.Ignore, unique_fields)
+            result = OnConflict.Ignore
+        elif update_conflicts:
+            self._check_on_conflicts_supported(OnConflict.Update, unique_fields)
+            if not update_fields:
+                raise IntegrityError(
+                    'You need to specify which fields you want to update'
+                )
+            for f in update_fields:
+                try:
+                    if f == 'pk':
+                        f = self.model._meta.pk.name
+                    self.model._meta.get_field(f)
+                except exceptions.FieldDoesNotExist:
+                    raise IntegrityError(
+                        'Field you specify is not in table: %s' % f
+                    )
+            result = OnConflict.Update
+        else:
+            result = None
+
+        return result
+
+    def bulk_create(
+        self, objs, batch_size=None, ignore_conflicts=False,
+        update_conflicts=False, update_fields=None, unique_fields=None
```

Comment:
Please add a trailing comma:
```suggestion
        update_conflicts=False, update_fields=None, unique_fields=None,
```